### PR TITLE
Crashlytics AW Sigaction crash fix #6434

### LINKED ISSUE #6434: Crashlytics crashes in WatchOS with sigaction on fatal signals is not supported
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.6
  * Firebase SDK version: 6.31.0
  * Firebase Component: Crashlytics
  * Component version: 4.5.0
  * Installation method: `CocoaPods`

### [REQUIRED] Step 2: Describe the problem

After issue #4558 was closed thanks to PR #6262 I was looking forward to use it in my WatchApp but it doesn't work.

The Watch App Crashes with the message `sigaction on fatal signals is not supported` in file [FIRCLSSignal.c:96](https://github.com/firebase/firebase-ios-sdk/blob/d2482a5a8bde9247c2f7a28027237b5d037189ba/Crashlytics/Crashlytics/Handlers/FIRCLSSignal.c#L96).

A quick google search for that message leads to this relevant part of [Apple's open source code](https://opensource.apple.com/source/Libc/Libc-1244.1.7/sys/sigaction.c) that suggest WatchOS doesn't support the use of `sigaction`

#### Steps to reproduce:

1. Create an iOS app and add a WatchOS App
2. Add Crashlytics to the Watch Extension target
3. Initialise Crashlytics in the `WKExtensionDelegate` applicationDidFinishLaunching
4. Run the app on a device without debugging

#### Relevant Code:

Cocoapods file:
```ruby

target 'WatchOSCrashlytics' do
  platform :ios, '11.0'
  use_frameworks!
  pod 'FirebaseCrashlytics', '~> 4.5.0'
end

target 'WatchApp Extension' do
  use_frameworks!
  platform :watchos, '6.0'
  pod 'FirebaseCrashlytics', '~> 4.5.0'
end
```

Extension delegate

```swift
import FirebaseCore

class ExtensionDelegate: NSObject, WKExtensionDelegate {

    func applicationDidFinishLaunching() {
        FirebaseConfiguration.shared.setLoggerLevel(.debug)
        FirebaseApp.configure();
    }
    ...
}
```

### PR DESCRIPTION
**Pull request by jostster**
This fixes the crashlytics crash on apple watch, which doesn't support sigaction. Fix #6434

### CODE REVIEW

**Comment by paulb777** [comment]
Thanks for the fix! Please fix the CI failures and add a note to https://github.com/firebase/firebase-ios-sdk/blob/master/Crashlytics/CHANGELOG.md.

**Comment by jostster** [comment]
@paulb777 looks like there _might_ be an issue with execution policy for tvos? Not able to see the details of why it is failing.

**Comment by paulb777** [comment]
@jostster Running `scripts/pod_lib_lint.rb FirebaseCrashlytics.podspec --platforms=tvos --verbose` in your branch shows

```
    Undefined symbols for architecture x86_64:
      "_FIRCLSSignalInitialize", referenced from:
          ___FIRCLSContextInitialize_block_invoke_2.29 in FirebaseCrashlytics(FIRCLSContext.o)
      "_FIRCLSSignalCheckHandlers", referenced from:
          ___52-[FIRCLSReportManager crashReportingSetupCompleted:]_block_invoke in FirebaseCrashlytics(FIRCLSReportManager.o)
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)


    Test session results, code coverage, and logs:
    	/Users/paulbeusterien/Library/Developer/Xcode/DerivedData/App-etocetmlokomdlfjvqpeqfycrpbt/Logs/Test/Run-FirebaseCrashlytics-Unit-unit-2020.09.10_10-22-02--0700.xcresult

    Testing failed:
    	Undefined symbol: _FIRCLSSignalInitialize
    	Undefined symbol: _FIRCLSSignalCheckHandlers
    	Testing cancelled because the build failed.
```

It looks like `#if` adjustments are needed.

**Review by paulb777 — COMMENTED** [review]
Thanks @jostster!  LGTM

Assigning to the Crashlytics team for review and approval.

File: Crashlytics/Crashlytics/Helpers/FIRCLSFeatures.h [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #define CLS_USE_SIGALTSTACK (!TARGET_OS_WATCH && !TARGET_OS_TV)
 #define CLS_CAN_SUSPEND_THREADS !TARGET_OS_WATCH
 #define CLS_MACH_EXCEPTION_SUPPORTED (!TARGET_OS_WATCH && !TARGET_OS_TV)
+#define CLS_SIGNAL_SUPPORTED !TARGET_OS_WATCH
```
Comment (samedson) [inline Crashlytics/Crashlytics/Helpers/FIRCLSFeatures.h:26]: Can you just add a comment saying "As of WatchOS 6 Signal crashes are not supported" and link to the original issue: https://github.com/firebase/firebase-ios-sdk/issues/6434

**Review by samedson — APPROVED** [review]
Just one ask for a comment - LGTM! 🚢

File: Crashlytics/Crashlytics/Helpers/FIRCLSFeatures.h (hunk 1 above)
Comment (jostster) [reply Crashlytics/Crashlytics/Helpers/FIRCLSFeatures.h]: @samedson  Done!

**Comment by jostster** [comment]
Comment added per review

**Comment by samedson** [comment]
Should be good to merge when you're ready @jostster 👍

**Comment by jostster** [comment]
@samedson I don't have write access so cannot merge.

**Comment by samedson** [comment]
Woops alright I got you!

**Pull request merged**